### PR TITLE
[Bitwarden] remove detailled view, fix login flow

### PR DESCRIPTION
**Pull request by pomdtr**
## Description

Fix an issue on the latest version of the bitwarden cli which consistentely autolocked the users vault.
I also removed the detailled view from the cli, it needs more work.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

File: extensions/bitwarden/src/search.tsx [hunk 1]
```diff
@@ -10,45 +10,65 @@ import {
...
+    },
+  };
+}
+
+export default function Search() {
+  const {
+    cliPath = process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw",
+    clientId,
+    clientSecret,
+  } = getPreferenceValues();
```
Comment (daniel-stoneuk) [inline extensions/bitwarden/src/search.tsx:52]:
```suggestion
  const { cliPath: cliPathPreference, clientId, clientSecret } = getPreferenceValues();
  const cliPath = cliPathPreference || process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw";
```

This destructuring doesn't handle when the preference is set to an empty string - i.e. been set and then removed in Raycast UI.

File: extensions/bitwarden/src/search.tsx [hunk 2]
```diff
@@ -147,94 +174,95 @@ function ItemListItem(props: {
...
+              />
+            ) : null}
+            {item.login?.totp ? (
+              <Action
+                shortcut={{ modifiers: ["cmd"], key: "t" }}
+                title="Copy TOTP"
+                icon={Icon.Clipboard}
+                onAction={() => {
+                  copyTotp(sessionToken, item.id);
+                  exit();
```
Comment (daniel-stoneuk) [inline extensions/bitwarden/src/search.tsx:197]: `copyTotp` contains code to `closeMainWindow`. Could remove it from there and keep the `exit()` here.

Comment (daniel-stoneuk) [reply extensions/bitwarden/src/search.tsx:197]: Nevermind, the `exit()` stops `copyTotp` from running here.

File: extensions/bitwarden/src/search.tsx [hunk 3]
```diff
@@ -86,11 +124,12 @@ function ItemList(props: {
...
             })
             .map((item) => (
-              <ItemListItem
+              <BitwardenItem
                 key={item.id}
                 item={item}
+                lockVault={session.deleteToken}
                 refreshItems={refreshItems}
-                sessionToken={sessionToken}
+                sessionToken={session.token}
```
Comment (daniel-stoneuk) [inline extensions/bitwarden/src/search.tsx:132]: Could we avoid passing two things down (`sessionToken` and `copyTotp`) to the `BitwardenItem` by wrapping the function definition in `useCallback`?

```typescript
  const copyTotp = useCallback(
    async (id: string) => {
      if (session.token) {
        const totp = await bitwardenApi.getTotp(id, session.token);
        Clipboard.copy(totp);
        exit();
      } else {
        showToast(Toast.Style.Failure, "Failed to fetch TOTP.");
      }
    },
    [session.token]
  );
```

We could also add a loading Toast while the command is running.

File: extensions/bitwarden/src/generate-password.tsx [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
 const GeneratePassword = () => {
-  const bitwardenApi = new Bitwarden();
+  const {
+    cliPath = process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw",
+    clientId,
+    clientSecret,
+  } = getPreferenceValues();
+  if (!existsSync(cliPath)) {
+    return <TroubleshootingGuide />;
+  }
```
Comment (daniel-stoneuk) [inline extensions/bitwarden/src/generate-password.tsx:31]: Same comment here as in `search.tsx`

**Review by daniel-stoneuk — COMMENTED** [review]
Just added a few comments from my testing. 

I've also noticed a regression from a fully logged out vault. To repro:

1) run `bw logout` in a terminal window.
2) run Search command and enter password. This fails.
3) enter password again. This succeeds.

File: extensions/bitwarden/src/search.tsx (hunk 1 above)
Comment (pomdtr) [reply extensions/bitwarden/src/search.tsx]: Good catch !

File: extensions/bitwarden/src/search.tsx (hunk 3 above)
Comment (pomdtr) [reply extensions/bitwarden/src/search.tsx]: Why is a useCallback necessary here ?

File: extensions/bitwarden/src/search.tsx (hunk 2 above)
Comment (pomdtr) [reply extensions/bitwarden/src/search.tsx]: Fixed !

File: extensions/bitwarden/src/search.tsx (hunk 3 above)
Comment (daniel-stoneuk) [reply extensions/bitwarden/src/search.tsx]: You're right, there's no need for useCallback here - sorry!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @pomdtr 👋 

Seems to work, I was one of them with the lock problem, and it works as expected now 💪 

Two minor things:

- Could be nice with a hotkey to lock the vault
- Could be nice with usage of [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) copy bitwarden logo to a new file with size 64x64

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Hi @pomdtr 👋 

Thanks for the update, it looks really good!

**Comment by daniel-stoneuk** [comment]
After using this PR for a little while, I've noticed the `popToRoot` call is a little inconvenient when copying both the username and password. 

What do you think?

**Comment by pomdtr** [comment]
> After using this PR for a little while, I've noticed the `popToRoot` call is a little inconvenient when copying both the username and password.
> 
> What do you think?

I agree, I just removed them.

**Pull request merged**